docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc

```diff
@@ -0,0 +1,92 @@
+= VSHN Managed Secret and PKI Management
+
+== Problem
+
+We need to provide a secret and PKI management solution on Kubernetes with the following features:
+
+* Secret storage with an API
+* PKI management with an API
+* Request and use secrets and certificates from this store in OpenShift on demand
+
+== Solutions
+
+There are currently multiple solutions available to satisfy above mentioned features:
+
+* https://www.vaultproject.io[Hashicorp Vault]
+* https://openbao.org[OpenBao]
+* https://github.com/Infisical/infisical[Infisical]
+
+All listed solutions meet the requirements.
+We therefore need to dig deeper to figure out which solution is the best for us.
+
+=== Analysis Vault
+
+Vault is the most popular and widespread solution on the market. It is probably the most feature-complete solution on the market as of now.
+However, it recently changed it's license to https://www.hashicorp.com/bsl[BSL]. This unfortunately limits how we can offer Vault to our customers.
+
+Advantages::
+
+* The de facto standard in the industry
+* Self-Hostable
+* Supported by many applications and libraries
+
+
+Disadvantages::
+
+* Not open-source
+* Self hosted solution is only available in the Enterprise tier. Pricing for that tier is unknown and requires contacting their sales team
+
+
+=== Analysis OpenBao
+
+OpenBao is a community-driven fork of Vault and managed by the Linux Foundation. It currently is compatible with Vault. However, this might change in the future, as Vault and OpenBao will develop independently.
+
+
+Advantages::
+
+* Open-Source
+* Self-Hostable
+* Free of charge
+* Compatible with the Vault API
+* Backed by the Linux Foundation
+
+Disadvantages::
+
+* Might deviate from Vault in the future and not be compatible anymore
+
+
+=== Analysis Infisical
+
+Infisical is an alternative solution that can also be self-hosted.
+It offers a free and paid version.
+
+
+Advantages::
+
+* Open-Source
+* Self-Hostable
+
+
+Disadvantages::
+
+* Many key features are not open-source and require the paid version:
+** Secret Versioning
+** RBAC
+** Secret Rotation
+** SAML SSO
+** IP Allowlisting
+
+* Not compatible with the Vault API
+
+== Decision
+
+We use OpenBao
```

Comment:
While I agree with going with OpenBao, it currently doesn't even have a release or a helm chart...

https://github.com/openbao/openbao

We should probably wait a while for this fork to get settled before actually using it.

Reply:
https://openbao.org/docs/platform/k8s/helm/

they expose link to `helm for kubernetes` but it throws 404 :D

Reply:
Yes I agree. They haven't ported the helm chart yet. We could probably use the official vault helm chart use the OpenBao image until they have their own chart ready.

Reply:
We could use the official vault helm chart and override the image until they have their own chart.